app/src/main/java/edu/grinnell/kdic/MainActivity.java

```diff
@@ -154,23 +143,11 @@ private void setupPlaybackToolbar() {
         View.OnClickListener onToggleVisualizeFragment = new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                if (!radioService.isLoading()) {
-                    if (mBackStack.peek() != R.id.visualizer) {
-                        showVisualizeFragment();
-                        mBackStack.add(R.id.visualizer);
-                    } else {
-                        hideVisualizeFragment();
-                        mBackStack.pop();
-                    }
-                    updateNavigationView();
-                }
+                setBottomNavigationOnClickListeners(v);
```

Comment:
I think the right method name for this, will be `toggleVisualizeFragmentOnClick(...)`